pylint/testutils/run.py

```diff
@@ -0,0 +1,46 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
```

Comment:
If we want to make all functions here private we could also consider renaming this file to `testutils/_run.py` and not adding anything from it to the `__all__` over `testutils`.

Reply:
Making everything explicitly private might end up having the opposite effect: makes people thing that if something is not prefixed by an underscore then it's public. We might want to list explicitly what is public API instead ?

Reply:
Isn't that we are assuming currently?

Reply:
Well it depends on our wet finger estimation of something's external use. For example we did not assume that the constant of the ``FormatChecker`` where public API in #6455 because it's probably not used but it can be imported and there is no underscore. But if we modify the BaseChecker or something really central then we deprecate.

Reply:
Hmm okay. I'd say let's try and do stuff as we would like them to be and worry less about potential implication for modules we haven't updated yet. I think underscoring files that are not public is good practice and also done by others such as `pytest`. If a file has mixed external and internal functions the internal ones should be underscored and the externals be added to the respective `__all__`. That way we document it both through the `__all_` and the use of underscores (which I think is often more visible to users).

Reply:
👍 Now that I think of it, the test ``Run`` is actually a nice utility if you need to run pylint tests with default values in a repository where there's a pylintrc, I wonder if we should make it public with the same interface than ``pylint.lint.Run`` ?

Reply:
The issue I have with this is: how much trouble do we save plugin writers compared to the trouble we get when we want to change anything related to these functions.
The functional test framework is a good example of a very useful test utility that would take a plugin writer considerable time to write and for which it would be good to get updates if we make any.
On the other hand, this utility is so straightforward that I wouldn't be surprised if plugin writers already wrote their own. It just adds overhead, maintenance and size to the overall package with relatively little benefit, imo.

That's also why I'm so "involved" in this particular discussion. I don't think this is such a valuable function that we should make it public while knowing that it is using arguments that we are in the process of deprecating (`do_exit`).
If this was my personal project I would likely add to this to a private `internal_testutils` package which wouldn't be included in the package at all.

Obviously I'm fine with putting it in if you want to, but then we should perhaps consider already removing the `do_exit` argument from `_Run`.

Reply:
No you're right let's make this file private.